### ISSUE TITLE
Release 0.8

### ISSUE THREAD

**Issue by starwed** [craftyjs/Crafty#1067]
I think we're at a point where it makes sense to do another release: there are at least two nice features in develop (layer and control improvements.)  Things I'd consider blockers:

- Some site upgrades are necessary, e.g. the doc generator needs a bit of work to make it clearer that you're looking at a system instead of a component.
- We probably want more testing; I'll create a testing release and link to it from the main site and google group.
- On that same note, we'll want to run the test suite and a few playgrounds in multiple browsers/OSes.

**Comment by starwed** [comment]
>at least two nice features in develop

Realized that there are actually several features added last spring that will be included in this release.  I still need to put together proper release notes and generate the api docs, but [here](https://github.com/craftyjs/Crafty/releases/tag/0.8.0-rc1) is the release candidate.

**Comment by starwed** [comment]
@mucaho  It looks like there's an issue with one of the testing-branch-only saucelabs tests that get triggered by travis.  ([Logged error here](https://travis-ci.org/craftyjs/Crafty/builds/187944569#L541)).  Do you know if this is a real test failure, or maybe just a configuration problem?  (The actual [sauce labs](https://saucelabs.com/u/mucaho) run associated with build 955 seems to be fine?)

**Comment by starwed** [comment]
Generated beta documentation along with some doc upgrades.  You can check it out [here](http://craftyjs.com/beta/api/).

(Most of the doc upgrades are about making it more obvious what a particular page/section is about -- adding "Component/System/Class" badges, to the headers, and marking methods in the `standard()` way.

**Comment by matthijsgroen** [comment]
Hey guys, I finally found some time again to work on my game, so obviously I'm going to look more at Crafty as well. I saw the following things when looking at the documentation:

- `Crafty.image_whitelist()` is in snake_case, while the rest is in lowerCamelCase.
- Maybe update the comment in `getVersion`?
- `Crafty.settings` does not state which settings are available, and what values are accepted

Is this version compatible with 0.7.1 ? if not, (doesn't need to be we are still in the 0. range) does it make sense to create an upgrade guide to help developers to the most notable changes?

**Comment by starwed** [comment]
>Crafty.image_whitelist() is in snake_case, while the rest is in lowerCamelCase.

Yeah, we can change that name and (I guess) leave some un-documented backwards compatible stubs.

>`Crafty.settings` does not state which settings are available, and what values are accepted

Yeah, it  has always been this way. :|  At some point we need to consolidate and document all the global options/settings (see #1026)

>Is this version compatible with 0.7.1 ?

I think upgrading will be easy (I call out the breaking changes I'm aware of in the [release notes](https://github.com/craftyjs/Crafty/wiki/Release-Notes)), but there are enough under-the-hood changes I figured it deserves the bump to 0.8.  People relying on being able to manually trigger keyboard events in multiway/etc would have to make some updates.

**Comment by matthijsgroen** [comment]
One thing I noticed breaking was centered text. The beta documentation states that `.css` properties are ignored when having a `Text` compontent, yet it advertises with `.css('textAlign', 'center')` (which is weird). I noticed that you can fix this by using `.textAlign`

https://github.com/craftyjs/Crafty/blob/develop/src/graphics/text.js#L267

But this method is nowhere in the API documentation.

**Comment by matthijsgroen** [comment]
On the page: http://craftyjs.com/beta/api/Crafty-createLayer.html

the examples use `Crafty.s`, but it looks like this should be `Crafty.createLayer` ?

**Comment by matthijsgroen** [comment]
Also question about the layer system, these have to be defined per-scene right? (and are cleaned up if the scene is destroyed?) (documentation doesn't tell anything about this)

**Comment by starwed** [comment]
>the examples use Crafty.s, but it looks like this should be Crafty.createLayer ?

Yup.  It also needs to discuss that you can access layers using `Crafty.s()`.  

>Also question about the layer system, these have to be defined per-scene right? (and are cleaned up if the scene is destroyed?) (documentation doesn't tell anything about this)

As currently implemented, custom layers should respond to scene changes just like regular layers -- meaning they will persist across scene changes.  I can make a note of that in the docs.

**Comment by starwed** [comment]
- Updated the release candidate with the recent round of fixes, and published the beta docs.
- Added an announcement to the landing page on craftyjs.com.

**Comment by starwed** [comment]
Poked around at the failing open sauce tests.
- Fixed most of the qunit tests, see #1075; once I got all the tests running there was a single test failure on iOS.  It seemed to be a safari bug related to requestAnimationFrame in some older versions of the ios simulator, so I resolved it by upgrading to 9.1.  AFAICT the bug probably is due to using it from a test harness, and usage older than 9.1 is less than 1% in any case, so we should be good there.
- That revealed failures in the second set of cloud tests -- I don't think these have ever actually successfully run before on the `testing` branch.  The errors seem to be related to fetching the wrong url, but I can't quite see what's causing them.  I'm comfortable ignoring them for this release.

*e:* Did fix the url issue, but that revealed another round of issues.   Although the screenshot comparison tests are quite valuable, they seem hard to get working consistently across all platforms.  :(

**Comment by matthijsgroen** [comment]
I noticed some more stuff missing in the documentation:

in the Gravity component:

```
 this.bind("LiftedOffGround", this._startGravity); // start gravity if we are off ground
        this.bind("LandedOnGround", this._stopGravity); // stop gravity once landed
```

There are events about wether a item has landed, but its not in the api documentation

**Comment by starwed** [comment]
>There are events about wether a item has landed, but its not in the api documentation

There are, but they're listed under [Supportable](http://craftyjs.com/beta/api/Supportable.html), which is the component that fires them.

The general pattern has been to document events where they are fired, rather than where they are consumed.  (It does seem like it would make sense to at least mention the latter!)

**Comment by starwed** [comment]
Got a working subset of the open-sauce tests working, and merged in to the testing branch.  (See PR #1078, and the followup issue #1079).

**Comment by matthijsgroen** [comment]
Another thing I found: http://craftyjs.com/beta/api/Crafty-scene.html
In documentation it refers to a css text-align, but this has to be `.textAlign` of the Text component

**Comment by starwed** [comment]
Hmm, text-align is also used as the main example in the `css()` method docs.

**Comment by starwed** [comment]
Fixed the text-align references for now -- there's a better fix that I'll land in develop post-release.

The saucelabs tests are pretty fragile; one times out on iOs about 1/3 of the time, and there was a problem with the chrome webdriver on the latest chrome beta, so I switched our test target to `latest` instead.  Eventually got them all to pass on the testing branch, so I'll go ahead and actually release 0.8 now!
